frame/election-provider-multi-phase/test-staking-e2e/src/mock.rs

```diff
@@ -594,18 +617,63 @@ pub fn roll_to(n: BlockNumber, delay_solution: bool) {
 	}
 }
 
+// Progress to given block, triggering session and era changes as we progress and ensuring that
+// there is a solution queued when expected.
+pub fn roll_to_with_ocw(n: BlockNumber, pool: Arc<RwLock<PoolState>>, delay_solution: bool) {
+	for b in (System::block_number()) + 1..=n {
+		System::set_block_number(b);
+		Session::on_initialize(b);
+		Timestamp::set_timestamp(System::block_number() * BLOCK_TIME + INIT_TIMESTAMP);
+
+		ElectionProviderMultiPhase::on_initialize(b);
+		ElectionProviderMultiPhase::offchain_worker(b);
+
+		if !delay_solution && pool.read().transactions.len() > 0 {
+			// build and submit extrinsics queued in the pool by ocw.
+			let encoded = pool.read().transactions[0].clone();
```

Comment:
Could there be more transactions in the pool for OCW? Would it be better to filter through the transactions to look for the one we need?

Reply:
Done!